docs/src/references/tedge-configuration-management.md

```diff
@@ -0,0 +1,365 @@
+sequenceDiagram
+  participant Mapper/others
+  participant Plugin
+  participant Tedge Agent
+
+  Mapper/others->>Plugin: tedge config_update command (Status: init)
+  Plugin->>Mapper/others: Status: executing
+  alt No error
+    Plugin->>Tedge Agent: File upload [HTTP]
+    Tedge Agent-->> Plugin: Status OK [HTTP]
+    Plugin->>Mapper/others: Status: successful
+  else Any error occurs
+    Plugin->>Mapper/others: Status: failed
+  end
+```
+
+## Handling config update commands
+
+By a config update operation, the plugin downloads a requested configuration file from the tedge file transfer repository,
+and locates it to the target path.
+
+The plugin subscribes to config update commands on the `<root>/<identifier>/cmd/config_update/+` MQTT topics.
+For example, it subscribes to the following topic for the main device.
+
+```sh te2mqtt
+tedge mqtt sub 'te/device/main///cmd/config_update/+'
+```
+
+A new config snapshot command with the ID "1234" is published by another component as below.
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/config_update/1234' '{
+  "status": "init",
+  "tedgeUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/config_update/mosquitto-1234",
+  "remoteUrl": "http://www.my.url",
+  "type": "mosquitto"
+}'
+```
+
+On reception of a configuration update command, the plugin:
+   1. `GET`s the content from the `tedgeUrl` specified by the command.
+   2. Uses the `type`(`mosquitto`) information to to look up the target path from `tedge-configuration-plugin.toml` file,
+   and applies the new configuration content to the corresponding `path`(`/etc/mosquitto/mosquitto.conf`).
+
+During the process, the plugin updates the command status via MQTT
+by publishing a retained message to the same `<root>/<identifier>/cmd/config_update/<id>` topic,
+where the command is received.
+The payload contains all the received data plus `path` information.
+
+On the reception of a new config update command, the plugin updates the status to `executing`.
+After all operation steps are successfully completed, the plugin updates the status to `successful`.
+If any unexpected error occurs, the plugin updates the status to `failed` with a `reason`.
+
+Thus, the operation status update message for the above example looks like below.
+
+```sh te2mqtt
+tedge mqtt pub -r 'te/device/main///cmd/config_update/1234' '{
+  "status": "successful",
+  "tedgeUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/config_update/mosquitto-1234",
+  "remoteUrl": "http://www.my.url",
+  "type": "mosquitto",
+  "path": "/etc/mosquitto/mosquitto.conf"
+}'
+```
+
+### Flow
+
+```mermaid
+sequenceDiagram
+  participant Mapper/others
+  participant Plugin
+  participant Tedge Agent
+
+  Mapper/others->>Plugin: tedge config_update command (Status: init)
+  Plugin->>Mapper/others: Status: executing
+  alt No error
+  Plugin->>Tedge Agent: Download a config file [HTTP]
+    Plugin->>Plugin: Apply the config
+    Plugin->>Tedge Agent: Delete a symlink [HTTP]
```

Comment:
Yeah makes sense. Let's push the responsibility to the mapper.

Reply:
We could also add a cache deletion mechanism in the future:
* delete files older than x days
* only keep last x files
* only keep up to x MB of data, deleting oldest first

Reply:
@reubenmiller jut FYI, unlike the `c8y-firmware-plugin`, there's no caching mechanism in the `c8y-config-plugin` currently. It directly downloads the config plugin from the cloud to the file transfer repo and deletes that file after a configuration is applied. But, while dealing with a fleet of child devices, it might make sense to cache these files as well, just like firmware plugin does. So, I'd be in favour of adding that caching functionality as well. We'd have to do add that to `tedge-agent` as well for software binaries for child devices.

Then the deletion of old/stale config, software and firmware files can be handled with some external mechanism as you suggested (even a cron job that runs periodically).